apps/apputil.cpp

```diff
@@ -420,16 +425,55 @@ string srt_json_cat_names [] = {
     "recv"
 };
 
+#ifdef HAS_PUT_TIME
+        // Follows ISO 8601
+std::string SrtStatsWriter::print_timestamp()
+{
+    using namespace std;
+    using namespace std::chrono;
+
+    const auto   systime_now = system_clock::now();
+    const time_t time_now    = system_clock::to_time_t(systime_now);
+
+    std::ostringstream output;
+
+    // SysLocalTime returns zeroed tm_now on failure, which is ok for put_time.
+    const tm tm_now = SysLocalTime(time_now);
+    output << std::put_time(&tm_now, "%FT%T.") << std::setfill('0') << std::setw(6);
+    const auto    since_epoch = systime_now.time_since_epoch();
+    const seconds s           = duration_cast<seconds>(since_epoch);
+    output << duration_cast<microseconds>(since_epoch - s).count();
+    output << std::put_time(&tm_now, "%z");
+    return output.str();
+}
+#else
+
+// This is a stub. The error when not defining it would be too
+// misleading, so this stub will work if someone mistakenly adds
+// the item to the output format without checking that HAS_PUT_TIME.
+string SrtStatsWriter::print_timestamp()
+{ return "<NOT IMPLEMENTED>"; }
```

Comment:
Maybe better to output some zero time, so that the output is still parsable?
E.g. `0000-00-00T00:00:00.000`

Reply:
This is only a stub for bugs in the software. This function simply should not be used if `HAS_PUT_TIME` is ndef. An alternative would be to not provide anything, but this would cause linker errors in this case that are more misleading and harder to resolve. OTOH maybe some better method could be used that would cause a compile error when this function is used.

Reply:
I missed that the absolute time point is not output in case `HAS_PUT_TIME` is not defined. Then ok.